client/lib/query-manager/media/test/index.js

```diff
@@ -0,0 +1,339 @@
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true for mime group wildcard match', () => {
+				const isMatch = manager.matches( {
+					mime_type: 'image/%'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+
+			it( 'should return false for mime subgroup partial wildcard match', () => {
+				const isMatch = manager.matches( {
+					mime_type: 'ima%/%'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return false for mime group partial wildcard match', () => {
+				const isMatch = manager.matches( {
+					mime_type: 'image/gi%'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+		} );
+
+		context( 'query.post_ID', () => {
+			it( 'should return false if post ID does not match', () => {
+				const isMatch = manager.matches( {
+					post_ID: 0
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true if post ID matches', () => {
+				const isMatch = manager.matches( {
+					post_ID: 41
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+		} );
+
+		context( 'query.before', () => {
+			it( 'should return false if query is not ISO 8601', () => {
+				const isMatch = manager.matches( {
+					before: '2018'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return false if media is not before date', () => {
+				const isMatch = manager.matches( {
+					before: '2010-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true if media is before date', () => {
+				const isMatch = manager.matches( {
+					before: '2018-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+		} );
+
+		context( 'query.after', () => {
+			it( 'should return false if query is not ISO 8601', () => {
+				const isMatch = manager.matches( {
+					after: '2010'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return false if media is not after date', () => {
+				const isMatch = manager.matches( {
+					after: '2018-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true if media is after date', () => {
+				const isMatch = manager.matches( {
+					after: '2010-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+		} );
+	} );
+
+	describe( '#sort()', () => {
+		context( 'query.order', () => {
+			it( 'should sort descending by default', () => {
+				const sorted = [
+					{ ID: 200 },
+					{ ID: 400 }
+				].sort( manager.sort.bind( manager, {
+					order_by: 'ID'
+				} ) );
+
+				expect( sorted ).to.eql( [
+					{ ID: 400 },
+					{ ID: 200 }
+				] );
+			} );
+
+			it( 'should reverse order when specified as ascending', () => {
+				const sorted = [
+					{ ID: 200 },
```

Comment:
Let's reverse the order in the input array, so we can verify the `ASC` sorting works as expected.